### ISSUE TITLE
MANPATH changed after mixing prepend_path and append_path

### ISSUE THREAD

**Issue by stierhofj** [TACC/Lmod#710]
Lmod version 8.7.43

This is related to https://github.com/TACC/Lmod/issues/590 and the other MANPATH issues.

Loading a modulefile with append_path or prepend_path adding to MANPATH where the empty entry (for system manual entries) is in the middle drops the empty entry. As a result, system entries are no longer accessible.
 
**To Reproduce**
Set the manpath to somethink like
`export MANPATH="/opt/sw1/man::/opt/sw2/man"`
This can be the result when two modules are loaded, one using prepend_path, the other append_path.
Load any (additional) module that appends/prepends to MANPATH

Resulting MANPATH is
`/opt/sw1/man:/opt/sw2/man:/opt/sw3/man` (append_path used here for example)

**Expected behavior**
Have the empty entry preserved in MANPATH
`/opt/sw1/man:/opt/sw2/man::/opt/sw3/man`

**Desktop (please complete the following information):**
 - OS: Linux
 - Linux distribution: Ubuntu-20.04
 - Lmod Version: 8.7.43
 - miniConfig:
  ```
  Modules based on Lua: Version 8.7.43 2024-06-24 15:50 -06:00
     by Robert McLay mclay@tacc.utexas.edu

  Changes from Default Configuration
  ----------------------------------

  Name                         Where Set  Default                                                  Value
  ----                         ---------  -------                                                  -----
  LFS_VERSION                  D          1.6.3                                                    1.8.0
  LMOD_ALLOW_TCL_MFILES        C          yes                                                      no
  LMOD_AVAIL_STYLE             E          <system>                                                 remeis
  LMOD_CACHED_LOADS            Other      no                                                       yes
  LMOD_FAST_TCL_INTERP         C          yes                                                      no
  LMOD_MODULEPATH_INIT         C          /data/software/Core/lmod/lmod/init/.modulespath          /data/software/.modulefiles/.modulepaths
  LMOD_PACKAGE_PATH            D          nil                                                      /data/software/.modulefiles
  LMOD_PAGER                   C          less                                                     /usr/bin/less
  LMOD_SITEPACKAGE_LOCATION    Other      /data/software/Core/lmod/8.7.43/libexec/SitePackage.lua  /data/software/.modulefiles/SitePackage.lua
  LMOD_SYSTEM_DEFAULT_MODULES  E          __unknown__                                              .system
  LMOD_TCLSH                   C          tclsh                                                    /usr/bin/tclsh
  LMOD_USING_FAST_TCL_INTERP   Other      yes                                                      no
  MODULEPATH_ROOT              E                                                                   /data/software/.modulefiles/
  PATH_TO_LUA                  C          lua                                                      /data/software/Core/lua/5.4.4/bin/lua


  Where Set -> D: default, E: environment, C: configuration
               lmod_cfg: lmod_config.lua SitePkg: SitePackage StdPkg: StandardPackage
               Other: Set somewhere outside of normal locations
```

**Comment by rtmclay** [comment]
Thanks very much for the bug report!  I was able to reproduce this issue and repair it for me.  If you get the chance, please use the Lmod testing branch to see if it fixes it for you.  Thanks again!

**Comment by stierhofj** [comment]
That was fast! It does now work as expected. Thanks for the quick fix!

**Comment by rtmclay** [comment]
This fix is now part of Lmod 8.7.44.  Closing this issue
